patches/minecraft/net/minecraft/world/raid/Raid.java.patch

```diff
@@ -0,0 +1,72 @@
+--- a/net/minecraft/world/raid/Raid.java
++++ b/net/minecraft/world/raid/Raid.java
+@@ -201,12 +201,14 @@
+ 
+    public void func_221295_o() {
+       this.field_221353_q = false;
++      if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.RaidEvent.Stop(this))) return;
+       this.field_221355_s.func_201360_b();
+       this.field_221360_x = Raid.Status.STOPPED;
+    }
+ 
+    public void func_221307_p() {
+       if (!this.func_221310_d()) {
++    	 if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.RaidEvent.Tick(this))) return;
+          if (this.field_221360_x == Raid.Status.ONGOING) {
```

Comment:
These tick events are highly costly, and should be designed to not need to be called every tick. What are you expecting to happen on people's implementations?
Perhaps a Wave start event or some system of queueing things to happen in the next tick would be a better approach?

Reply:
will address this the next time i try a raid event pr